server/aesdsocket.c

```diff
@@ -367,90 +506,124 @@ void exit_cleanup()
 
 /**
  * @brief   Called when SIGINT or SIGTERM are received.
- * 
- * @param   none
- * 
+ *
+ * @param   void
+ *
  * @return  void
-*/
-void sig_int_term_handler()
+ */
+void sig_int_term_handler(void)
 {
     printf("Exiting...\n");
     syslog(LOG_INFO, "Exiting...\n");
-    exit_cleanup();
-    exit(EXIT_SUCCESS);
+    sig_exit_status = 1;
```

Comment:
@ajaykandagal nice change here, note that you need similar strategies for `sig_alarm_handler` though.